src/auth/envDetect.ts

```diff
@@ -62,6 +65,15 @@ function isCloudFunction() {
   return !!(process.env.FUNCTION_NAME || process.env.FUNCTION_TARGET);
 }
 
+/**
+ * This check only verifies that the environment is running knative.
+ * This must be run *after* checking for Kubernetes, otherwise it will
+ * return a false positive.
+ */
+function isCloudRun() {
+  return !!process.env.K_CONFIGURATION;
+}
```

Comment:
I guess it's too late now but we should be querying metadata endpoint first and use this env variable only in case metadata endpoint doesn't return the environment